### ISSUE TITLE
context.channelRelativeUrl (and possible others) are not set on mobile clients

### ISSUE THREAD

**Issue by davelosert** [OfficeDev/microsoft-teams-library-js#286]
I was just debugging my app within the mobile client (Android 9 with Samsung Galaxy S9) and realized that not all context-variables are set there.

In my case, the `channelRelativeUrl` variable is missing which is crucial for my app.

Within the desktop app as well as within the browser, it is always set.
Is this a bug or is there a reason why it is not available in mobile?

**Comment by Gousia-MSFT** [comment]
@davelosert Are you trying this from Bot or Tab?

**Comment by davelosert** [comment]
@Gousia-MSFT : Thanks for your answer. 👍 

I am trying this from a Tab.
I am implementing an image-gallery that shows the images of a channel's associated sharepoint folder (the same files which are shown in the `files` tab) in a grid-gallery.

I am trying to get the channels `driveItems` using the GraphAPI by constructing an URL with a relative file path out of the context as followed:

```JavaScript
microsoftTeams.getContext(context => {
    // this throws on mobile as channelRelativeUrl is not defined
    const channelFolderArray = context.channelRelativeUrl.split('/');
    const channelFolder = channelFolderArray[channelFolderArray.length - 1];
    const url = `https://graph.microsoft.com/v1.0/groups/${context.groupId}/drive/root:/${channelFolder}:/children?$expand=thumbnails`;
    callGraphAPI(url);
})
```

As an alternative, I thought I might be able to use the `channelName` to construct this URL - but the `channelName` sometimes differs from the actualy `folderName` in sharepoint (e.g. in translations or when the channel got renamed).

Do you have any other workaround for this? 


_PS: I know that just yesterday, an advanced files-tab was announced and will be released which kind of includes the grid view of files. I'd still like to finish the app as it is more for learning purposes :)_

**Comment by davelosert** [comment]
Just realized that there was a bug ticket for this already (but it got closed) so I am linking it: #245

**Comment by Gousia-MSFT** [comment]
@davelosert We do have a bug for this which is being tracked internally. Thanks!
